int64[] centerX
int64[] centerY
